packages/shared/src/components/CombatReport/CombatVideo/index.tsx

```diff
@@ -62,46 +76,135 @@ export const CombatVideo = () => {
+    (e: MouseEvent<HTMLInputElement>) => {
+      if (!vidRef.current) return;
+      if (!rangeRef.current) return;
+      vidRef.current.currentTime = parseFloat(e.currentTarget.value) + matchStartTime;
+    },
+    [matchStartTime],
+  );
+
+  const rangeBarDrag = useCallback(
+    (e: MouseEvent<HTMLInputElement>) => {
+      if (!vidRef.current) return;
+      if (!rangeRef.current) return;
+      if (!(e.buttons === 1)) return;
+      vidRef.current.currentTime = parseFloat(e.currentTarget.value) + matchStartTime;
+    },
+    [matchStartTime],
+  );
+
+  useEffect(() => {
+    const controller = new AbortController();
+    const signal = controller.signal;
+    if (vidRef.current && rangeRef.current) {
+      vidRef.current.addEventListener(
+        'timeupdate',
+        () => {
+          if (!rangeRef.current) return;
+          if (!vidRef.current) return;
+          rangeRef.current.value = `${vidRef.current.currentTime - matchStartTime}`;
+        },
+        {
+          signal,
+        },
+      );
+    }
+
+    return () => {
+      controller.abort();
+    };
+  }, [combat?.durationInSeconds, matchStartTime, videoInformation]);
 
   if (errorMessage) {
     return <div className="animate-fadein flex flex-col gap-2">{errorMessage}</div>;
   }
-  if (!combat || !foundVodRef) {
+  if (!combat || !videoInformation) {
     return null;
   }
 
   return (
     <div className="animate-fadein flex flex-col gap-2 flex-1 h-full">
-      <button
-        className="btn"
-        onClick={() => {
-          if (!vidRef.current) return;
-          if (!foundVodRef.metadata) return;
-
-          const offset = getMatchTimeoffsetSeconds(combat.id, foundVodRef.metadata);
-          vidRef.current.currentTime = offset || MATCH_START_CORRECTION;
-          vidRef.current.play();
-        }}
-      >
-        Play from start of round
-      </button>
-      <video
-        controls
-        id="video"
-        ref={vidRef}
-        // domain here looks weird but chrome will canonicalize the item in the string it thinks is the domain
-        // which will lead to the b64 string losing its casing :(
-        src={`vod://wowarenalogs/${btoa(foundVodRef.videoPath)}`}
-        style={{
-          maxHeight: 'calc(100vh - 200px)', // TODO: MIGHTFIX figure out how to contain video without allowing scrollbars here
-        }}
-      />
+      <div className="flex flex-row gap-2">
+        <div className="flex flex-col gap-2">
+          <button
+            className="btn"
+            onClick={() => {
+              if (!vidRef.current) return;
+              if (!videoInformation.metadata) return;
+
+              const offset = getMatchTimeoffsetSeconds(
+                combat.id,
+                videoInformation.metadata,
+                videoInformation.compensationTimeSeconds || 0,
+              );
+              vidRef.current.currentTime = offset || 0;
+              vidRef.current.play();
+            }}
+          >
+            Start
```

Comment:
would be most ideal to have these buttons somewhere close to wherever the time slider is.